### DISCUSSION TITLE
rnote-cli without local build

### DISCUSSION THREAD

**Discussion by Hiqqup** [flxzt/rnote#786] [Q&A]
Is it somehow possible to use the cli crate without needing to build locally and install all the dependencies? It would be a gamechanger for me because I like having my files in svg and the cli can convert it and i can automate that. Dont spare me with the technical details on why if so its not possible, i am really curious how it works.

**Comment by flxzt** [comment]
Like shipping a built cli binary with every release? It would probably be possible.

If you only want to build the cli without the ui, you can skip installing the `gtk4`, `libadwaita` and `poppler` dependencies. It is then possible to build the cli through `meson compile cli-cargo-build -C  _mesonbuild`. The binary is then located in `_mesonbuild/rnote-cli`. The engine (the core code of the app that is shared between the cli and ui) searches for some installed files but it can handle the case when they are not found.

**Reply by retrop7** [reply]
Just want to second this.  Command line interface with the release would be really useful to me - though same as @Hiqqup - I'm also looking for a way to automate converting to SVG so if there was a check box in the GUI to automatically backup as SVG I'd be happy with that also.

**Reply by TimerErTim** [reply]
Late to the party but I "third" this. Having the cli be easily downloadable from github instead of setting up the whole build chain would be so practical. I can imagine rnote could then even be added to mise-en-place, which would make document generation easily automatable.

**Reply by flxzt** [reply]
you are able to execute the rnote-cli through `flatpak run --command=rnote-cli com.github.flxzt.rnote`
